api/webapp.go

```diff
@@ -59,15 +61,57 @@ func (s Server) FrontendHandler(prefix string) http.Handler {
 	// TODO -
 	// * there's no error handling,
```

Comment:
I think we need to handle errors here. We can do it by making `updates` a channel of error instead of boolean values. starting on line 65:
```go
     updating := true
     var fetchError error
     errs := make(chan error)
	go func() {
		if err := s.resolveWebappPath(); err != nil {
	          errs <- err
                  return
		}
		path := s.Config().Webapp.EntrypointHash

		// TODO (ramfox): frontend handler should accept a context
		log.Info("fetching webapp off the distributed web...")
		fetcher, ok := s.Repo().Store().(cafs.Fetcher)
		if !ok {
			errs <- fmt.Error("this store cannot fetch from remote sources")
			return
		}
		if _, err := fetcher.Fetch(cafs.SourceAny, path+"/main.js"); err != nil {
			errs <- fmt.Errorf("error fetching file: %s", err.Error())
			return
		}
		log.Info("pinning webapp...")
		pinner, ok := s.Repo().Store().(cafs.Pinner)
		if !ok {
			errs <- fmt.Error("unable to pin webapp")
			return
		}
		if err := pinner.Pin(path, true); err != nil {
			errs <- fmt.Errorf("error pinning webapp: %s", err.Error())
			return
		}
		log.Info("done pinning webapp!")
		errs <- nil
	}()

	go func() {
	   fetchErr = <-errs
           if fetchErr != nil {
               log.Errorf("error fetching webapp:  %s", fetchErr)
           }
          updating = false
	}()
```

Also, I think we should avoid references to 'your repository'. It might be the user's repository, but we can't always assume as much (example: when tests are running, it's not really "your" repo).